docs/DevGuide/Travis.md

```diff
@@ -0,0 +1,88 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+
+# Travis CI {#travis_guide}
+
+SpECTRE uses
+[TravisCI](https://travis-ci.org/sxs-collaboration/spectre) for
+testing the code.  Multiple build jobs (described below) are launched
+each time a pull request is submitted or updated.  In addition, a
+daily cron job is run.  Travis will also launch these build jobs each
+time you push to a branch on your fork of SpECTRE.  For pull requests,
+you can see view the Travis CI build by clicking on `Details` next to
+the `Travis build failed` or `Travis build succeeded` in the section
+for `All checks have failed` or `All checks have passed` towards the
+bottom of the pull request.  You can login to Travis using your GitHub
+credentials in order to cancel your builds and enable additional
+options.
+
+## What is tested
+
+The Travis report lists the build jobs which will each have either a
+green check mark if it passes, or a red `X` if it has failed.
+Clicking on a build job will display the log of the build.
+
+The following build jobs are launched:
+* CHECK_COMMITS runs the script `tools/CheckCommits.sh` and fails the build if
+any casing of the words in the list below is the first word of the commit
+message.  This allows developers to flag their commits with these keywords to
+indicate that a pull request should not be merged in its current state.
+  - fixup
+  - wip (for work in progress)
+  - fixme
+  - deleteme
+  - rebaseme
+  - testing
+  - rebase
+* CHECK_FILES runs the script `tools/CheckFiles.sh` (which also runs the script
+`tools/FileTestDefs.sh`) and fails the build if any of the following checks are
+true:
+  - A `c++` file (i.e. `*.hpp` or `*.cpp` file) contains a line over 80
```

Comment:
This should also include `*.tpp`, and I will submit a pull request momentarily to make that actually true.